src/app/integrations/list-page/list-page.component.ts

```diff
@@ -29,12 +37,54 @@ export class IntegrationsListPage implements OnInit {
     ],
   }];
 
-  constructor(private store: IntegrationStore) {
+  constructor(
+    private store: IntegrationStore,
+    private modalService: ModalService,
+    private integrationSupportService: IntegrationSupportService,
+    public notificationService: NotificationService,
+    public application: ApplicationRef,
+    private http: Http,
+    private config: ConfigService,
+  ) {
     this.integrations = this.store.list;
     this.loading = this.store.loading;
+    const url = this.config.getSettings().apiEndpoint + '/integration-support/import';
```

Comment:
This block also feels like it should go into the integration-support service instead of here.  Also there's already a restangular service that you can use to build this URL...